python-frontend/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -560,21 +560,32 @@ private void addSymbolUsage(Name nameTree) {
     }
 
     private void addTypeToSymbol(Name nameTree, Expression rhs) {
-      Type type = null;
-      if (rhs.is(Kind.CALL_EXPR)) {
-        Symbol calleeSymbol = ((CallExpression) rhs).calleeSymbol();
-        if (calleeSymbol != null && calleeSymbol.kind() == Symbol.Kind.CLASS) {
-          ClassSymbol classSymbol = (ClassSymbol) calleeSymbol;
-          if (classSymbol.superClasses().isEmpty() && !classSymbol.hasUnresolvedTypeHierarchy()) {
-            type = new Type(calleeSymbol);
-          }
-        }
-      }
+      Type type = rhs.is(Kind.CALL_EXPR) ? getReturnType((CallExpression) rhs) : null;
       SymbolImpl symbol = currentScope().resolve(nameTree.name());
       if (symbol != null && symbol.usages().stream().filter(Usage::isBindingUsage).count() == 1) {
         symbol.setType(type);
       }
     }
 
+    @CheckForNull
+    private Type getReturnType(CallExpression rhs) {
+      Symbol calleeSymbol = rhs.calleeSymbol();
+      if (calleeSymbol == null) {
+        return null;
+      }
+      if (calleeSymbol.kind() == Symbol.Kind.CLASS) {
+        ClassSymbol classSymbol = (ClassSymbol) calleeSymbol;
+        if (classSymbol.superClasses().isEmpty() && !classSymbol.hasUnresolvedTypeHierarchy()) {
```

Comment:
I know it's not linked to this PR but do you think we could add a comment to explain why we have this condition `classSymbol.superClasses().isEmpty()` ? As it should eventually be replaced, I believe it's good to keep it in mind.